src/core/monitors/JointConditionalAncestralStateMonitor.h

```diff
@@ -151,26 +153,30 @@ void JointConditionalAncestralStateMonitor<characterType>::monitorVariables(unsi
     }
     else
     {
-        dist_bd = dynamic_cast<StateDependentSpeciationExtinctionProcess*>( &nodes[0]->getDistribution() ); // this does!
+        dist_bd = dynamic_cast<StateDependentSpeciationExtinctionProcess*>( &cdbdp->getDistribution() ); // this does!
+        if ( dist_bd == NULL )
+        {
+        	dist_glhbdsp = dynamic_cast<GeneralizedLineageHeterogeneousBirthDeathSamplingProcess*>( &cdbdp->getDistribution() );
+        }
         num_sites = 1;
         num_nodes = tree->getValue().getNumberOfNodes();
     }
         
-    std::vector<std::vector<characterType> > startStates(num_nodes,std::vector<characterType>(num_sites));
-    std::vector<std::vector<characterType> > endStates(num_nodes,std::vector<characterType>(num_sites));
+    std::vector<std::vector<characterType> > startStates(num_nodes, std::vector<characterType>(num_sites));
+    std::vector<std::vector<characterType> > endStates(num_nodes, std::vector<characterType>(num_sites));
     
     // draw ancestral states
     if ( ctmc != NULL )
     {
         dist_ctmc->drawJointConditionalAncestralStates(startStates, endStates);
     }
-    else
+    else if ( dist_bd != NULL )
```

Comment:
I'm curious why you don't just write `else if ( dist_bd )`, since null pointers convert to false, and non-null pointers convert to true.